# Cooking similarity - études et exploitation des données (1)

Ce notebook explique les différentes étapes de tris et extraction du jeu de données <font color=blue>"full_format_recipes.json"</font>.

Il s'agit d'un jeu de données de 20 000 recettes extraites du site [Epicurious](https://www.epicurious.com/).

Ce jeu de données nous permet de travailler sur des recettes diverses avec plusieurs informations tel que les valeurs nutritionnelles, la  note de la recette, les ingrédients, catégories...

## Source
Jeu de données obtenu sur: https://www.kaggle.com/hugodarwood/epirecipes

## Contenu

Plus de 20 000 recettes listées selon l’évaluation de la recette, l’information nutritionnelle et la(les) catégorie(s) attribuée(s) etc...

In [4]:
import numpy as np
import pandas as pd 
from pandas import read_csv

df = pd.read_json("full_format_recipes.json")

In [5]:
df.head()

calories                                         categories  \
0     426.0  [Sandwich, Bean, Fruit, Tomato, turkey, Vegeta...   
1     403.0  [Food Processor, Onion, Pork, Bake, Bastille D...   
2     165.0  [Soup/Stew, Dairy, Potato, Vegetable, Fennel, ...   
3       NaN  [Fish, Olive, Tomato, Sauté, Low Fat, Low Cal,...   
4     547.0  [Cheese, Dairy, Pasta, Vegetable, Side, Bake, ...   

                 date                                               desc  \
0 2006-09-01 04:00:00                                               None   
1 2004-08-20 04:00:00  This uses the same ingredients found in boudin...   
2 2004-08-20 04:00:00                                               None   
3 2009-03-27 04:00:00  The Sicilian-style tomato sauce has tons of Me...   
4 2004-08-20 04:00:00                                               None   

                                          directions   fat  \
0  [1. Place the stock, lentils, celery, carrot, ...   7.0   
1  [Combine first 9 ingredients in heavy medium s...  23.0   
2  [In a large heavy saucepan cook diced fennel a...   7.0   
3  [Heat oil in heavy large skillet over medium-h...   NaN   
4  [Preheat oven to 350°F. Lightly grease 8x8x2-i...  32.0   

                                         ingredients  protein  rating  sodium  \
0  [4 cups low-sodium vegetable or chicken stock,...     30.0   2.500   559.0   
1  [1 1/2 cups whipping cream, 2 medium onions, c...     18.0   4.375  1439.0   
2  [1 fennel bulb (sometimes called anise), stalk...      6.0   3.750   165.0   
3  [2 tablespoons extra-virgin olive oil, 1 cup c...      NaN   5.000     NaN   
4  [1 12-ounce package frozen spinach soufflé, th...     20.0   3.125   452.0   

                                         title  
0              Lentil, Apple, and Turkey Wrap   
1  Boudin Blanc Terrine with Red Onion Confit   
2                Potato and Fennel Soup Hodge   
3             Mahi-Mahi in Tomato Olive Sauce   
4                    Spinach Noodle Casserole

## Tris des données

Nous avons trié les données par catégories pour obtenir des jeux de données plus petits afin de les exploiter et les afficher dans notre interface web. 
Les jeux de données doivent contenir des catégories de recette pertinente tel que "Tarte", "Pâte" ou "Cookie" de sorte à montrer une certaine diversité dans les ingrédients, les directions...

Voici le détail de notre analyse afin de générer les différents jeu de données:

***
Trier les recettes par catégories:

In [6]:
#Nombre de résultats par catégorie
def getNumbersOfRecipeByCategories(df):
    result = {}
    
    for categoriesList in df.categories:
        
        # check if it is really a list of categories
        if not isinstance(categoriesList, list):
            # if not, skip
            continue
            
        for categorie in categoriesList:
            if categorie in result:
                result[categorie] += 1
            else:
                result[categorie] = 1
    return result

In [7]:
getNumbersOfRecipeByCategories(df)

{'Sandwich': 428,
 'Bean': 577,
 'Fruit': 1958,
 'Tomato': 2140,
 'turkey': 456,
 'Vegetable': 2087,
 'Kid-Friendly': 1791,
 'Apple': 666,
 'Lentil': 98,
 'Lettuce': 225,
 'Cookie': 69,
 'Food Processor': 973,
 'Onion': 2238,
 'Pork': 928,
 'Bake': 4413,
 'Bastille Day': 30,
 "New Year's Eve": 165,
 'Dried Fruit': 206,
 'Port': 110,
 'Winter': 3099,
 'Chill': 1126,
 'Bon Appétit': 9355,
 'Soup/Stew': 1164,
 'Dairy': 1496,
 'Potato': 1128,
 'Fennel': 371,
 'Gourmet': 6648,
 'New York': 49,
 'Fish': 936,
 'Olive': 556,
 'Sauté': 2044,
 'Low Fat': 929,
 'Low Cal': 1049,
 'High Fiber': 1013,
 'Dinner': 2705,
 'Healthy': 2351,
 'Simmer': 890,
 'Pescatarian': 6042,
 'Dairy Free': 3206,
 'Peanut Free': 8390,
 'Tree Nut Free': 7044,
 'Soy Free': 8088,
 'Kosher': 6175,
 'Cheese': 1355,
 'Pasta': 968,
 'Side': 3151,
 'Vegetarian': 6846,
 'Quick & Easy': 5372,
 'Fall': 3015,
 'California': 146,
 'Basil': 565,
 'Bacon': 587,
 'Summer': 4151,
 'Salad': 1516,
 'Easter': 234,
 'Ham': 222,
 'Asparagus

La liste ci-dessus nous donne le nombre de recettes par catégories dans le désordre. On peut voir des catégories qui sont moins pertinentes tel que des ingrédients comme "Onion","Tomato"... ou qui possède très peu de résultats (<200).
***
Pour obtenir la catégorie contenant le plus de recettes:

In [8]:
import operator
recipeCategCount = getNumbersOfRecipeByCategories(df)
max(recipeCategCount, key=recipeCategCount.get)

'Bon Appétit'

On remarque que la catégorie possédant le plus de recette est "Bon Appétit". Cette catégorie n'est pas pertinente car elle peut rassembler plusieurs recettes différentes qui n'ont aucune similarité comme on peut le voir sur le site d'Epicurious en allant dans cette catégorie:

<img src="categorie_bonappetit.png">

Le but est donc de trouver des recettes qui ont des bases similaires mais qui présentent aussi des variantes.


***
Trier les catégories par ordre décroissant de nombre de recettes:

In [10]:
#sorted_recipeCategCount = sorted(recipeCategCount.items(), key=operator.itemgetter(1))
sorted_recipeCategCount = sorted(recipeCategCount.items(), key=operator.itemgetter(1), reverse=True)
sorted_recipeCategCount

[('Bon Appétit', 9355),
 ('Peanut Free', 8390),
 ('Soy Free', 8088),
 ('Tree Nut Free', 7044),
 ('Vegetarian', 6846),
 ('Gourmet', 6648),
 ('Kosher', 6175),
 ('Pescatarian', 6042),
 ('Quick & Easy', 5372),
 ('Wheat/Gluten-Free', 4906),
 ('Bake', 4413),
 ('Summer', 4151),
 ('Dessert', 3573),
 ('Dairy Free', 3206),
 ('Side', 3151),
 ('No Sugar Added', 3132),
 ('Winter', 3099),
 ('Fall', 3015),
 ('Dinner', 2705),
 ('Sugar Conscious', 2466),
 ('Healthy', 2351),
 ('Kidney Friendly', 2313),
 ('Onion', 2238),
 ('Tomato', 2140),
 ('Vegetable', 2087),
 ('Sauté', 2044),
 ('Milk/Cream', 1995),
 ('Fruit', 1958),
 ('Vegan', 1851),
 ('Kid-Friendly', 1791),
 ('Egg', 1768),
 ('Spring', 1715),
 ('Herb', 1681),
 ('Garlic', 1643),
 ('Salad', 1516),
 ('Dairy', 1496),
 ('Thanksgiving', 1458),
 ('Appetizer', 1372),
 ('Lunch', 1359),
 ('Cheese', 1355),
 ('Chicken', 1344),
 ('Roast', 1320),
 ('No-Cook', 1251),
 ('Soup/Stew', 1164),
 ('Cocktail Party', 1154),
 ('Ginger', 1146),
 ('Potato', 1128),
 ('Chill', 11

En regardant la liste des catégories triées on peut ainsi dégager des catégories interessantes tel que:
1. ('Pasta', 968) avec 968 recettes de pâtes.
2. ('Sandwich', 428) avec 428 recettes de sandwich.
3. ('Cookies', 273) avec 273 recettes de cookies.
Nous allons donc ensuite générer les jeu de données en json pour ces 3 catégories:
***
Génération des fichier json:

In [11]:
def keepCategorie(df, categorie):
    # There is probably a better way to do this...
    rowToDelete = []
    for row, categories in enumerate(df.categories):
        if not isinstance(categories, list):
            rowToDelete.append(row)
        elif not categorie in categories:
                rowToDelete.append(row)

    df = df.drop(rowToDelete)
    return df

### Génération du fichier "Pasta.json".

In [34]:
# On veut garder seulement les pasta
df_pasta = keepCategorie(df,'Pasta')

# Avec le reset_index(drop=True) on reinitialise les index à partir de 0.
df_pasta.reset_index(drop=True).to_json('Pasta.json',orient='records')
pd.read_json("Pasta.json")
#head_pasta = pd.read_json("Pasta.json")
#head_pasta.head()

calories                                         categories  \
0       547.0  [Cheese, Dairy, Pasta, Vegetable, Side, Bake, ...   
1         NaN  [Milk/Cream, Cheese, Leafy Green, Pasta, Bake,...   
2         NaN  [Cheese, Herb, Pasta, Low Cal, Dinner, Lunch, ...   
3       634.0  [Fish, Leafy Green, Pasta, Tomato, Bake, Quick...   
4       684.0  [Cheese, Dairy, Garlic, Herb, Pasta, Vegetable...   
5       209.0  [Garlic, Olive, Pasta, Side, Quick & Easy, Gou...   
6         NaN  [Pasta, Tomato, Bake, Cod, Bell Pepper, Chard,...   
7       724.0  [Ginger, Herb, Nut, Pasta, Soy, Sauté, Vegetar...   
8         NaN  [Olive, Pasta, Tomato, Vegetarian, Quick & Eas...   
9         NaN  [Onion, Pasta, Sauté, Low Fat, Quick & Easy, R...   
10        NaN  [Garlic, Pasta, Sauté, Vegetarian, Kid-Friendl...   
11      303.0  [Garlic, Ginger, Nut, Pasta, Vegetable, Sauté,...   
12      375.0  [Beef, Mushroom, Onion, Pasta, Tomato, Sauté, ...   
13      200.0  [Pasta, Side, Quick & Easy, Lunch, Lemon, Lime...   
14      272.0  [Soup/Stew, Pasta, Quick & Easy, Lunch, Boil, ...   
15      787.0  [Milk/Cream, Dairy, Mushroom, Pasta, Pork, Sau...   
16        NaN  [Bean, Pasta, Low Fat, High Fiber, Spinach, Ch...   
17     1370.0  [Nut, Pasta, Pepper, Sauté, Vegetarian, Pine N...   
18      414.0  [Egg, Pasta, Brunch, Bake, Vegetarian, Quick &...   
19      910.0  [Garlic, Pasta, Sauté, Vegetarian, Quick & Eas...   
20      394.0  [Citrus, Nut, Pasta, Sauté, Vegetarian, Quick ...   
21      487.0  [Salad, Pasta, Tomato, Side, Picnic, Vegetaria...   
22        NaN  [Cheese, Lamb, Pasta, Quick & Easy, High Fiber...   
23        NaN  [Chicken, Leafy Green, Mushroom, Pasta, Poultr...   
24      603.0  [Pasta, Tomato, Kid-Friendly, Dinner, Lunch, S...   
25      537.0  [Milk/Cream, Pasta, Christmas, Vegetarian, Qui...   
26      353.0  [Milk/Cream, Dairy, Herb, Mushroom, Pasta, Sau...   
27      333.0  [Salad, Pasta, Pepper, Appetizer, Side, Super ...   
28        NaN  [Soup/Stew, Wine, Egg, Ginger, Pasta, Quick & ...   
29     1699.0  [Milk/Cream, Chicken, Pasta, Braise, High Fibe...   
..        ...                                                ...   
938       NaN  [Milk/Cream, Fish, Pasta, Easter, Quick & Easy...   
939      60.0  [Milk/Cream, Dairy, Pasta, Side, Bake, Vegetar...   
940       NaN  [Cheese, Chicken, Dairy, Leafy Green, Nut, Pas...   
941    1058.0  [Chicken, Dairy, Pasta, Poultry, Vegetable, Ki...   
942     520.0  [Cheese, Herb, Nut, Pasta, Sauté, Vegetarian, ...   
943       NaN  [Salad, Mushroom, Pasta, Vegetable, Low Fat, V...   
944     726.0  [Cheese, Pasta, Vegetable, Spring, Prosciutto,...   
945     567.0  [3-Ingredient Recipes, Pasta, Dinner, Pumpkin,...   
946     434.0  [Salad, Olive, Pasta, Quick & Easy, Buffet, Mo...   
947     589.0  [Dairy, Leafy Green, Pasta, Sauté, Quick & Eas...   
948     396.0  [Soup/Stew, Pasta, Shellfish, Tomato, Sauté, C...   
949     689.0  [Milk/Cream, Mushroom, Onion, Pasta, Bake, Cas...   
950     291.0  [Milk/Cream, Fish, Pasta, Potato, Bake, Cod, F...   
951     338.0  [Salad, Cheese, Citrus, Pasta, Tomato, Quick &...   
952       NaN  [Pasta, Dinner, Lunch, Broccoli Rabe, Bon Appé...   
953     139.0  [Soup/Stew, Pasta, Tomato, turkey, Lunch, Fall...   
954    3842.0  [Pasta, Tomato, Bake, Kid-Friendly, Dinner, Mo...   
955       NaN  [Cheese, Pasta, Shellfish, Sauté, Valentine's ...   
956       NaN  [Herb, Pasta, Side, Vegetarian, Peanut, Bon Ap...   
957     498.0  [Salad, Fish, Pasta, Quick & Easy, Tuna, Chill...   
958     537.0  [Dairy, Pasta, Side, Bake, Vegetarian, Cassero...   
959     566.0  [Soup/Stew, Bean, Pasta, Tomato, Sauté, Parmes...   
960     659.0  [Pasta, Tomato, Quick & Easy, Basil, Tuna, Sum...   
961     727.0  [Garlic, Pasta, Kid-Friendly, Quick & Easy, Di...   
962     666.0  [Pasta, Quick & Easy, Parmesan, Lemon, Arugula...   
963    1127.0  [Pasta, Pork, Tomato, Kid-Friendly, Dinner, Gr...   
964     750.0  [Citrus, Fish, Garlic, Herb, Pasta, Pepper, Ve

### Génération du fichier "Sandwich.json"

In [33]:
# On veut garder seulement les sandwich
df_sw = keepCategorie(df,'Sandwich')
df_sw.reset_index(drop=True).to_json('Sandwich.json',orient='records')
pd.read_json("Sandwich.json")
#head_sw = pd.read_json("Sandwich.json")
#head_sw.head()

calories                                         categories  \
0       426.0  [Sandwich, Bean, Fruit, Tomato, turkey, Vegeta...   
1       948.0  [Sandwich, Food Processor, Tomato, Kid-Friendl...   
2       904.0  [Sandwich, Beef, Cheese, Mushroom, Mustard, Qu...   
3       641.0  [Sandwich, Beef, Cheese, Leafy Green, Onion, Q...   
4       959.0  [30 Days of Groceries, Chickpea, Salad, Sandwi...   
5       352.0  [Salad, Sandwich, Vegetarian, Kid-Friendly, Qu...   
6       374.0  [Sandwich, Bean, Egg, Kid-Friendly, Quick & Ea...   
7       245.0  [Sandwich, Beef, Dinner, Lunch, Meat, Ground B...   
8         NaN  [Sandwich, No-Cook, Kid-Friendly, Back to Scho...   
9         NaN  [Sandwich, Garlic, Onion, turkey, Summer, Gril...   
10      752.0  [Sandwich, Bourbon, Pork, Sauté, Super Bowl, D...   
11      654.0  [Sandwich, Ginger, Onion, Tomato, turkey, Back...   
12     1145.0  [Sandwich, Cheese, Onion, Vegetarian, Quick & ...   
13        NaN  [Sandwich, Onion, Pepper, Pork, Sausage, Bell ...   
14        NaN  [Sandwich, Onion, Tuna, Bell Pepper, Summer, T...   
15      144.0  [Sandwich, No-Cook, Vegetarian, Quick & Easy, ...   
16     1080.0  [Sandwich, Beef, Egg, Brie, Bell Pepper, Summe...   
17     1405.0  [Sandwich, Blender, Food Processor, Onion, tur...   
18      194.0  [Sandwich, Tomato, Appetizer, Kid-Friendly, Qu...   
19      538.0  [Sandwich, Herb, Lamb, Quick & Easy, Yogurt, S...   
20      555.0  [Sandwich, Milk/Cream, Tomato, turkey, Kentuck...   
21        NaN  [Sandwich, Cheese, Egg, Brunch, Quick & Easy, ...   
22       54.0  [Bread, Sandwich, Bake, Vegetarian, Basil, Rad...   
23      681.0  [Sandwich, Fish, Ginger, Vegetable, Cocktail P...   
24        NaN  [Sandwich, No-Cook, Vegetarian, Kid-Friendly, ...   
25      576.0  [Sandwich, Beef, Quick & Easy, Lunch, Ground B...   
26      646.0  [Sandwich, Fruit, Garlic, Leafy Green, Pork, T...   
27        NaN  [Sandwich, Onion, Quick & Easy, High Fiber, Lu...   
28        NaN  [Sandwich, Berry, Citrus, Onion, Poultry, turk...   
29       96.0  [Sandwich, Tomato, Cocktail Party, Kid-Friendl...   
..        ...                                                ...   
398     596.0  [Sandwich, Cheese, Mustard, turkey, Quick & Ea...   
399       NaN  [Sandwich, turkey, Vegetable, Quick & Easy, Mi...   
400     378.0  [Sandwich, Food Processor, Vegetarian, Rosemar...   
401      97.0  [Bon Appétit, Thanksgiving, turkey, Cornmeal, ...   
402       NaN  [Sandwich, Beef, Cheese, Dairy, Kid-Friendly, ...   
403    2168.0  [Sandwich, Chicken, Dairy, Herb, Poultry, Mari...   
404     857.0  [Sandwich, Beef, Leafy Green, Herb, Vegetable,...   
405       NaN  [Sandwich, turkey, Kid-Friendly, Low Cal, Dinn...   
406     623.0  [Sandwich, Mustard, Broil, Quick & Easy, Lunch...   
407       NaN  [Sandwich, Tea, Chicken, Poultry, Back to Scho...   
408     785.0  [Sandwich, Shellfish, Picnic, Quick & Easy, Di...   
409     678.0  [Sandwich, Pork, Fourth of July, Summer, Grill...   
410       NaN  [Sandwich, Onion, Tomato, turkey, Kid-Friendly...   
411       NaN  [Bread, Sandwich, Tomato, Picnic, Vegetarian, ...   
412     536.0  [Sandwich, Cheese, Super Bowl, Kid-Friendly, Q...   
413    1444.0  [Sandwich, Food Processor, Bean, Herb, Fry, Sa...   
414     821.0  [Sandwich, Vegetable, Broil, Quick & Easy, Lun...   
415     793.0  [Grill/Barbecue, Cabbage, Vegan, Flaming Hot S...   
416       NaN  [Sandwich, Cheese, Olive, Pork, Lunch, Celery,...   
417     301.0  [Sandwich, Citrus, Dairy, Quick & Easy, Lemon,...   
418     403.0  [Sandwich, Fish, Leafy Green, No-Cook, Picnic,...   
419       NaN  [Sandwich, Cheese, Egg, Breakfast, Brunch, Kid...   
420    1094.0  [Sandwich, Kid-Friendly, Quick & Easy, Apple, ...   
421       NaN  [Sandwich, turkey, Low Fat, Grill/Barbecue, Self]   
422     100.0  [Sandwich, Beef, Ground Beef, Summer, House & ...   
423       NaN  [Sandwich, Wine, Lamb, Tomato, Fourth of July,...   
424     443.0  [Sandwich, Broil, Ham, Swiss Cheese, Sour Crea

### Génération du fichier "Cookie.json"

In [32]:
# On veut garder seulement les cookies
df_cookie = keepCategorie(df,'Cookies')
df_cookie.reset_index(drop=True).to_json('Cookies.json',orient='records')
pd.read_json("Cookies.json")

calories                                         categories  \
0       129.0  [Cookies, Rum, Cheese, Dairy, Fruit, Dessert, ...   
1       105.0  [Cookies, Food Processor, Fruit, Dessert, Bake...   
2        87.0  [Cookies, Dessert, Bake, Christmas, Kid-Friend...   
3        92.0  [Cookies, Nut, Dessert, Bake, Christmas, Kid-F...   
4        76.0  [Bon Appétit, Bake, Cookies, Almond, Lemon, De...   
5       305.0  [Cookies, Cheese, Dairy, Fruit, Nut, Dessert, ...   
6       123.0  [Cookies, Passover, Kosher for Passover, Almon...   
7       300.0  [Cookies, Mixer, Dessert, Freeze/Chill, Christ...   
8        13.0  [Cookies, Mixer, Dessert, Bake, Freeze/Chill, ...   
9       223.0  [Cookies, Food Processor, Mixer, Dairy, Nut, D...   
10      297.0  [Cookies, Mixer, Nut, Vegetable, Dessert, Bake...   
11       18.0  [Cookies, Food Processor, Mixer, Nut, Bake, Ch...   
12      136.0  [Cookies, Dessert, Bake, Christmas, Edible Gif...   
13       52.0  [Cookies, Egg, Dessert, Bake, Lime, Spice, Gou...   
14      135.0  [Cookies, Mixer, Dessert, Bake, Kid-Friendly, ...   
15       88.0  [Cookies, Mixer, Dessert, Bake, Kid-Friendly, ...   
16      223.0  [Cookies, Food Processor, Mixer, Dairy, Nut, D...   
17      312.0  [Bon Appétit, Cookies, Bake, Chocolate, Coconu...   
18       50.0  [Cookies, Mixer, Egg, Ginger, Dessert, Bake, C...   
19      228.0  [Cookies, Chocolate, Ginger, Dessert, Bake, Su...   
20       81.0  [Cookies, Coffee, Blender, Chocolate, Bake, Ea...   
21       86.0  [Cookies, Bake, Lemon, Almond, Gourmet, Minnes...   
22      162.0  [Cookies, Chocolate, Fruit, Nut, Bake, Christm...   
23       93.0  [Cookies, Dessert, Bake, Orange, Shower, Honey...   
24      302.0  [Cookies, Coffee, Mixer, Egg, Dessert, Bake, C...   
25      134.0  [Cookies, Chocolate, Citrus, Ginger, Dessert, ...   
26        NaN   [Cookies, Chocolate, No-Cook, Christmas, Parade]   
27      106.0  [Cookies, Fruit, Nut, Dessert, Christmas, Drie...   
28        NaN  [Cookies, Nut, Bake, Christmas, Tree Nut, Almo...   
29       69.0  [Cookies, Fruit Juice, Liqueur, Citrus, Nut, B...   
..        ...                                                ...   
243      93.0  [Cookies, Food Processor, Nut, Dessert, Bake, ...   
244     139.0  [Dessert, Chocolate, Sesame, Cinnamon, Orange,...   
245       NaN  [Cookies, Nut, Bake, Christmas, Tree Nut, Almo...   
246      97.0  [Cookies, Blender, Mixer, Dessert, Bake, Date,...   
247       NaN  [Cookies, Food Processor, Mixer, Chocolate, Da...   
248      77.0  [Cookies, Ginger, Dessert, Bake, Christmas, Ve...   
249       NaN  [Cookies, Chocolate, Nut, Dessert, Bake, Chris...   
250     199.0  [Cookies, Dessert, Bake, Cherry, Oat, Kidney F...   
251     300.0  [Cookies, Mixer, Dessert, Freeze/Chill, Christ...   
252      92.0  [Cookies, Egg, Nut, Dessert, Bake, Lemon, Almo...   
253     219.0  [Cookies, Mixer, Cheese, Bake, Freeze/Chill, C...   
254       NaN  [Cookies, Mixer, Fruit, Nut, Bake, Christmas, ...   
255     250.0  [Cookies, Food Processor, Dessert, Bake, Freez...   
256       NaN  [Cookies, Berry, Dessert, Bake, Christmas, Win...   
257       NaN  [Cookies, Mixer, Egg, Breakfast, Bake, Brandy,...   
258      58.0  [Cookies, Mixer, Chocolate, Egg, Dessert, Bake...   
259      86.0  [Cookies, Mixer, Dessert, Bake, Kid-Friendly, ...   
260     310.0            [Cookies, Dessert, Butter, Bon Appétit]   
261      14.0  [Cookies, Ginger, Bake, Christmas, Almond, Win...   
262      83.0  [Cookies, Chocolate, Bake, Wheat/Gluten-Free, ...   
263      64.0  [Cookies, Chocolate, Dairy, Ginger, Dessert, B...   
264     106.0  [Cookies, Cheese, Dairy, Fruit, Tomato, Bake, ...   
265      43.0  [Cookies, Chocolate, Dessert, Bake, Christmas,...   
266      38.0  [Cookies, Mixer, Chocolate, Dessert, Bake, Ora...   
267      77.0                   [Cookies, Bake, Lemon, Hazelnut]   
268      74.0  [Cookies, Mixer, Fruit, Nut, Dessert, Bake, Fr...   
269     191.0  [Bon Appétit, Dessert, Christmas, Kid-Friendly

Afin de tester avec le minimum de données (et pas un fichier json de plus de 200 à 900 résultats...)nous avons créé un fichier "Test" avec quelques données:

In [31]:
#Générer json avec 5 recettes
head_cookie = pd.read_json("Cookies.json")
head_cookie.head()
head_cookie.to_json('Cookies_small.json',orient='records')
data_small = pd.read_json("Cookies_small.json")
data_small

calories                                         categories  \
0       129.0  [Cookies, Rum, Cheese, Dairy, Fruit, Dessert, ...   
1       105.0  [Cookies, Food Processor, Fruit, Dessert, Bake...   
2        87.0  [Cookies, Dessert, Bake, Christmas, Kid-Friend...   
3        92.0  [Cookies, Nut, Dessert, Bake, Christmas, Kid-F...   
4        76.0  [Bon Appétit, Bake, Cookies, Almond, Lemon, De...   
5       305.0  [Cookies, Cheese, Dairy, Fruit, Nut, Dessert, ...   
6       123.0  [Cookies, Passover, Kosher for Passover, Almon...   
7       300.0  [Cookies, Mixer, Dessert, Freeze/Chill, Christ...   
8        13.0  [Cookies, Mixer, Dessert, Bake, Freeze/Chill, ...   
9       223.0  [Cookies, Food Processor, Mixer, Dairy, Nut, D...   
10      297.0  [Cookies, Mixer, Nut, Vegetable, Dessert, Bake...   
11       18.0  [Cookies, Food Processor, Mixer, Nut, Bake, Ch...   
12      136.0  [Cookies, Dessert, Bake, Christmas, Edible Gif...   
13       52.0  [Cookies, Egg, Dessert, Bake, Lime, Spice, Gou...   
14      135.0  [Cookies, Mixer, Dessert, Bake, Kid-Friendly, ...   
15       88.0  [Cookies, Mixer, Dessert, Bake, Kid-Friendly, ...   
16      223.0  [Cookies, Food Processor, Mixer, Dairy, Nut, D...   
17      312.0  [Bon Appétit, Cookies, Bake, Chocolate, Coconu...   
18       50.0  [Cookies, Mixer, Egg, Ginger, Dessert, Bake, C...   
19      228.0  [Cookies, Chocolate, Ginger, Dessert, Bake, Su...   
20       81.0  [Cookies, Coffee, Blender, Chocolate, Bake, Ea...   
21       86.0  [Cookies, Bake, Lemon, Almond, Gourmet, Minnes...   
22      162.0  [Cookies, Chocolate, Fruit, Nut, Bake, Christm...   
23       93.0  [Cookies, Dessert, Bake, Orange, Shower, Honey...   
24      302.0  [Cookies, Coffee, Mixer, Egg, Dessert, Bake, C...   
25      134.0  [Cookies, Chocolate, Citrus, Ginger, Dessert, ...   
26        NaN   [Cookies, Chocolate, No-Cook, Christmas, Parade]   
27      106.0  [Cookies, Fruit, Nut, Dessert, Christmas, Drie...   
28        NaN  [Cookies, Nut, Bake, Christmas, Tree Nut, Almo...   
29       69.0  [Cookies, Fruit Juice, Liqueur, Citrus, Nut, B...   
..        ...                                                ...   
243      93.0  [Cookies, Food Processor, Nut, Dessert, Bake, ...   
244     139.0  [Dessert, Chocolate, Sesame, Cinnamon, Orange,...   
245       NaN  [Cookies, Nut, Bake, Christmas, Tree Nut, Almo...   
246      97.0  [Cookies, Blender, Mixer, Dessert, Bake, Date,...   
247       NaN  [Cookies, Food Processor, Mixer, Chocolate, Da...   
248      77.0  [Cookies, Ginger, Dessert, Bake, Christmas, Ve...   
249       NaN  [Cookies, Chocolate, Nut, Dessert, Bake, Chris...   
250     199.0  [Cookies, Dessert, Bake, Cherry, Oat, Kidney F...   
251     300.0  [Cookies, Mixer, Dessert, Freeze/Chill, Christ...   
252      92.0  [Cookies, Egg, Nut, Dessert, Bake, Lemon, Almo...   
253     219.0  [Cookies, Mixer, Cheese, Bake, Freeze/Chill, C...   
254       NaN  [Cookies, Mixer, Fruit, Nut, Bake, Christmas, ...   
255     250.0  [Cookies, Food Processor, Dessert, Bake, Freez...   
256       NaN  [Cookies, Berry, Dessert, Bake, Christmas, Win...   
257       NaN  [Cookies, Mixer, Egg, Breakfast, Bake, Brandy,...   
258      58.0  [Cookies, Mixer, Chocolate, Egg, Dessert, Bake...   
259      86.0  [Cookies, Mixer, Dessert, Bake, Kid-Friendly, ...   
260     310.0            [Cookies, Dessert, Butter, Bon Appétit]   
261      14.0  [Cookies, Ginger, Bake, Christmas, Almond, Win...   
262      83.0  [Cookies, Chocolate, Bake, Wheat/Gluten-Free, ...   
263      64.0  [Cookies, Chocolate, Dairy, Ginger, Dessert, B...   
264     106.0  [Cookies, Cheese, Dairy, Fruit, Tomato, Bake, ...   
265      43.0  [Cookies, Chocolate, Dessert, Bake, Christmas,...   
266      38.0  [Cookies, Mixer, Chocolate, Dessert, Bake, Ora...   
267      77.0                   [Cookies, Bake, Lemon, Hazelnut]   
268      74.0  [Cookies, Mixer, Fruit, Nut, Dessert, Bake, Fr...   
269     191.0  [Bon Appétit, Dessert, Christmas, Kid-Friendly

### Remarque:

En regardant les résultats des fichier générés on remarque que les intitulés des recettes correspondent.

Par exemple si on prend cette ligne de "Cookies.json":
<img src="ligne_roastedalmond.png">

et qu'on vérifie sur le site d'épicurious:

<img src="roasted_almond.png">
